homeassistant/helpers/translation.py

```diff
@@ -321,3 +330,47 @@ async def async_get_translations(
     for entry in cached:
         result.update(entry)
     return result
+
+
+@bind_hass
+def get_cached_translations(
+    hass: HomeAssistant,
+    language: str,
+    category: str,
+    integrations: Iterable[str] | None = None,
```

Comment:
As I understand it, we only need this for a single domain, which would allow us to greatly simplify the function?
```suggestion
    integration: str
```

Reply:
My goal was to keep make it similar to `async_get_translations`. It would be nice to reuse some code between them

Reply:
I have kept this approach to extract and reuse some common logic